includes/Abilities/BlockAbilities.php

```
@@ -1802,6 +1802,88 @@ private static function flatten_blocks_for_response(
+	 *
+	 * Walks through the updates array and validates any insert-child, replace-block,
+	 * or wrap-in-group operations against BlockContentPolicy. Returns null on success
+	 * (all operations pass policy), or a WP_Error on the first violation.
+	 *
+	 * This is called before apply_batch() to ensure all-or-nothing rejection.
+	 *
+	 * @param array<int,mixed> $updates Array of update specs from handle_update_blocks.
+	 * @return null|\WP_Error null on success, WP_Error on first policy violation.
+	 */
+	private static function preflight_tier_policy( array $updates ): ?\WP_Error {
+		foreach ( $updates as $idx => $update ) {
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$idx'. (undefined)

(UnusedLocalVariable)